standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -922,51 +928,91 @@ private void updateReplId(Connection dbConn, ReplLastIdInfo replLastIdInfo) thro
       rs = stmt.executeQuery("select PARAM_VALUE from TABLE_PARAMS where PARAM_KEY = " +
               "'repl.last.id' and TBL_ID = " + tblId);
       if (rs == null || !rs.next()) {
-        stmt.executeUpdate("insert into TABLE_PARAMS values ( " + tblId +
-                " , 'repl.last.id' , " + lastReplId + ")");
+        query = "insert into TABLE_PARAMS values ( " + tblId + " , 'repl.last.id' , ? )";
       } else {
-        stmt.executeUpdate("update TABLE_PARAMS set PARAM_VALUE = " + lastReplId + " where TBL_ID = " + tblId +
-                " and PARAM_KEY = 'repl.last.id'");
+        query = "update TABLE_PARAMS set PARAM_VALUE = ? where TBL_ID = " + tblId +
+                " and PARAM_KEY = 'repl.last.id'";
+      }
+
+      params = Arrays.asList(lastReplId);
+      pst = sqlGenerator.prepareStmtWithParameters(dbConn, query, params);
+      LOG.debug("Updating repl id for table <" + query.replaceAll("\\?", "{}") + ">", lastReplId);
+      if (pst.executeUpdate() != 1) {
+        //only one row insert or update should happen
+        throw new RuntimeException("TABLE_PARAMS is corrupted for table " + table);
       }
 
       if (partList == null || partList.isEmpty()) {
         return;
       }
 
-      for (String part : partList) {
-        query = "select \"PART_ID\" from \"PARTITIONS\" where \"TBL_ID\" = " + tblId + " and " + "\"PART_NAME\" = ? ";
-        params = Arrays.asList(part);
-        pst = sqlGenerator.prepareStmtWithParameters(dbConn, query, params);
-        LOG.debug("Going to execute query <" + query.replaceAll("\\?", "{}") + ">", quoteString(part));
+      List<String> questions = new ArrayList<>();
+      for(int i = 0; i < partList.size(); ++i) {
+        questions.add("?");
+      }
 
-        rs = pst.executeQuery();
-        if (rs == null || !rs.next()) {
-          throw new MetaException(" partition with name " + part + " does not exist in table " +
+      List<String> queries = new ArrayList<>();
+      StringBuilder prefix = new StringBuilder();
+      StringBuilder suffix = new StringBuilder();
+      prefix.append("select \"PART_ID\" from \"PARTITIONS\" where \"TBL_ID\" = " + tblId + " and ");
+
+      // Populate the complete query with provided prefix and suffix
+      List<Integer> counts = TxnUtils.buildQueryWithINClauseStrings(conf, queries, prefix, suffix,
+              questions, "\"PART_NAME\"", true, false);
+      int totalCount = 0;
+      for (int i = 0; i < queries.size(); i++) {
+        query = queries.get(i);
+        int partCount = counts.get(i);
+
+        LOG.debug("Going to execute query <" + query.replaceAll("\\?", "{}") + ">",
+                partList.subList(totalCount, totalCount+partCount));
+        pst = dbConn.prepareStatement(query);
+        for (int j = 0; j < partCount; j++) {
+          pst.setString(j + 1, partList.get(totalCount + j));
+        }
+        totalCount += partCount;
+        prs = pst.executeQuery();
```

Comment:
Need to close pre too. Not sure if it is auto closeable.

Reply:
prs  will be closed along with pst